beacon-chain/rpc/eth/config/handlers.go

```diff
@@ -124,11 +124,27 @@ func convertValueForJSON(v reflect.Value, tag string) interface{} {
 			if !v.Field(i).CanInterface() {
 				continue // unexported
 			}
-			key := f.Tag.Get("json")
-			if key == "" || key == "-" {
+			jsonTag := f.Tag.Get("json")
+			if jsonTag == "-" {
+				continue // skip fields with json:"-"
+			}
+
+			// Parse JSON tag options (e.g., "fieldname,omitempty")
+			parts := strings.Split(jsonTag, ",")
+			key := parts[0]
+			hasOmitEmpty := len(parts) > 1 && strings.Contains(strings.Join(parts[1:], ","), "omitempty")
```

Comment:
do we need to join the parts again?

```suggestion
			hasOmitEmpty := len(parts) > 1 && strings.Contains(jsonTag, "omitempty")
```